sw/device/lib/dif/dif_kmac_unittest.cc

```diff
@@ -147,10 +162,364 @@ class KmacTest : public testing::Test, public mock_mmio::MmioTest {
     // Check if there still unaligned data in the buffer tail.
     if (remaining) {
       // write unaligned data in bytes.
-      setExpectedMessageByte(message, remaining);
+      ExpectMessageByte(message, remaining);
+    }
+  }
+
+  void ExpectConfig(void) {
+    EXPECT_WRITE32_SHADOWED(
+        KMAC_CFG_SHADOWED_REG_OFFSET,
+        {{KMAC_CFG_SHADOWED_KMAC_EN_BIT, config_.enable},
+         {KMAC_CFG_SHADOWED_KSTRENGTH_OFFSET, config_.keyStrength},
+         {KMAC_CFG_SHADOWED_MODE_OFFSET, config_.mode},
+         {KMAC_CFG_SHADOWED_MSG_ENDIANNESS_BIT, config_.msgBigEndian},
+         {KMAC_CFG_SHADOWED_STATE_ENDIANNESS_BIT, config_.stateBigEndian},
+         {KMAC_CFG_SHADOWED_SIDELOAD_BIT, config_.sideload},
+         {KMAC_CFG_SHADOWED_ENTROPY_MODE_OFFSET, config_.entropyMode},
+         {KMAC_CFG_SHADOWED_ENTROPY_FAST_PROCESS_BIT,
+          config_.entropyFastProcess},
+         {KMAC_CFG_SHADOWED_MSG_MASK_BIT, config_.msg_mask},
+         {KMAC_CFG_SHADOWED_ENTROPY_READY_BIT, config_.entropyReady},
+         {KMAC_CFG_SHADOWED_ERR_PROCESSED_BIT, config_.errProcessed},
+         {KMAC_CFG_SHADOWED_EN_UNSUPPORTED_MODESTRENGTH_BIT,
+          config_.enableUnsupportedModeStrength}});
+  }
+
+  void ExpectKey(const dif_kmac_key_t &key) {
+    std::map<dif_kmac_key_length_t, uint32_t> keySize = {
+        {kDifKmacKeyLen128, KMAC_KEY_LEN_LEN_VALUE_KEY128},
+        {kDifKmacKeyLen192, KMAC_KEY_LEN_LEN_VALUE_KEY192},
+        {kDifKmacKeyLen256, KMAC_KEY_LEN_LEN_VALUE_KEY256},
+        {kDifKmacKeyLen384, KMAC_KEY_LEN_LEN_VALUE_KEY384},
+        {kDifKmacKeyLen512, KMAC_KEY_LEN_LEN_VALUE_KEY512},
+    };
+
+    EXPECT_WRITE32(KMAC_KEY_LEN_REG_OFFSET, keySize[key.length]);
+    for (uint32_t i = 0; i < ARRAYSIZE(key.share0); ++i) {
+      ptrdiff_t offset = KMAC_KEY_SHARE0_0_REG_OFFSET + (i * sizeof(uint32_t));
+      EXPECT_WRITE32(offset, key.share0[i]);
+      offset = KMAC_KEY_SHARE1_0_REG_OFFSET + (i * sizeof(uint32_t));
+      EXPECT_WRITE32(offset, key.share1[i]);
+    }
+  }
+
+  void ExpectPrefix(const uint32_t *prefix_regs, uint32_t size) {
+    for (uint32_t i = 0; i < size; ++i) {
+      ptrdiff_t offset = KMAC_PREFIX_0_REG_OFFSET + i * sizeof(uint32_t);
+      EXPECT_WRITE32(offset, prefix_regs[i]);
+    }
+  }
+};
+
+class Kmac256Test : public KmacTest {
+ protected:
+  dif_kmac_key_t key_ = {
+                .share0 = {0x43424140, 0x47464544, 0x4B4A4948, 0x4F4E4D4C,
+                           0x53525150, 0x57565554, 0x5B5A5958, 0x5F5E5D5C},
+                .share1 = {0},
+                .length = kDifKmacKeyLen256,
+            };
+  dif_kmac_mode_kmac_t mode_ = kDifKmacModeKmacLen256;
+  dif_kmac_customization_string_t custom_string_;
+
+  Kmac256Test() {
+    const std::string kString = "My Tagged Application";
```

Comment:
I think I said this elsewhere but it got dropped... `kConst` is only for constants baked into the program. kConst globals aren't really something we do.